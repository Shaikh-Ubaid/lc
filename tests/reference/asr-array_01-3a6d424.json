{
    "basename": "asr-array_01-3a6d424",
    "cmd": "lc --show-asr --no-color {infile} -o {outfile} -extra-arg=\"\"",
    "infile": "tests/../integration_tests/array_01.cpp",
    "infile_hash": "deb860e61919d1bd2fbc7abeeed41124193f4284de0838d4ba94fa09",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-array_01-3a6d424.stdout",
    "stdout_hash": "337b11bab91bf64e841d76a62768573e77f7b5b2db440edef9ad10e7",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}