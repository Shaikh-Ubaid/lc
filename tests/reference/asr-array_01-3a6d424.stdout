(TranslationUnit
    (SymbolTable
        1
        {
            main:
                (Function
                    (SymbolTable
                        2
                        {
                            __return_var:
                                (Variable
                                    2
                                    __return_var
                                    []
                                    ReturnVar
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            arr1d:
                                (Variable
                                    2
                                    arr1d
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Array
                                        (Integer 4)
                                        [((IntegerConstant 0 (Integer 4))
                                        (IntegerConstant 5 (Integer 4)))]
                                        FixedSizeArray
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            arr2d:
                                (Variable
                                    2
                                    arr2d
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Array
                                        (Integer 4)
                                        [((IntegerConstant 0 (Integer 4))
                                        (IntegerConstant 4 (Integer 4)))
                                        ((IntegerConstant 0 (Integer 4))
                                        (IntegerConstant 3 (Integer 4)))
                                        ((IntegerConstant 0 (Integer 4))
                                        (IntegerConstant 2 (Integer 4)))]
                                        FixedSizeArray
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            i:
                                (Variable
                                    2
                                    i
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            j:
                                (Variable
                                    2
                                    j
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            k:
                                (Variable
                                    2
                                    k
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            l:
                                (Variable
                                    2
                                    l
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            sum1d:
                                (Variable
                                    2
                                    sum1d
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            sum2d:
                                (Variable
                                    2
                                    sum2d
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    main
                    (FunctionType
                        []
                        (Integer 4)
                        Source
                        Implementation
                        ()
                        .false.
                        .false.
                        .false.
                        .false.
                        .false.
                        []
                        .false.
                    )
                    []
                    []
                    [(=
                        (Var 2 arr1d)
                        (ArrayConstant
                            [(IntegerConstant 1 (Integer 4))
                            (IntegerConstant 2 (Integer 4))
                            (IntegerConstant 3 (Integer 4))
                            (IntegerConstant 4 (Integer 4))
                            (IntegerConstant 5 (Integer 4))]
                            (Array
                                (Integer 4)
                                [((IntegerConstant 0 (Integer 4))
                                (IntegerConstant 5 (Integer 4)))]
                                FixedSizeArray
                            )
                            RowMajor
                        )
                        ()
                    )
                    (=
                        (Var 2 arr2d)
                        (ArrayReshape
                            (ArrayConstant
                                [(IntegerConstant 1 (Integer 4))
                                (IntegerConstant 2 (Integer 4))
                                (IntegerConstant 3 (Integer 4))
                                (IntegerConstant 4 (Integer 4))
                                (IntegerConstant 5 (Integer 4))
                                (IntegerConstant 6 (Integer 4))
                                (IntegerConstant 8 (Integer 4))
                                (IntegerConstant 9 (Integer 4))
                                (IntegerConstant 10 (Integer 4))
                                (IntegerConstant 11 (Integer 4))
                                (IntegerConstant 12 (Integer 4))
                                (IntegerConstant 13 (Integer 4))
                                (IntegerConstant 14 (Integer 4))
                                (IntegerConstant 15 (Integer 4))
                                (IntegerConstant 16 (Integer 4))
                                (IntegerConstant 17 (Integer 4))
                                (IntegerConstant 18 (Integer 4))
                                (IntegerConstant 19 (Integer 4))
                                (IntegerConstant 20 (Integer 4))
                                (IntegerConstant 21 (Integer 4))
                                (IntegerConstant 22 (Integer 4))
                                (IntegerConstant 23 (Integer 4))
                                (IntegerConstant 24 (Integer 4))
                                (IntegerConstant 25 (Integer 4))]
                                (Array
                                    (Integer 4)
                                    [((IntegerConstant 0 (Integer 4))
                                    (IntegerConstant 24 (Integer 4)))]
                                    FixedSizeArray
                                )
                                RowMajor
                            )
                            (ArrayPhysicalCast
                                (ArrayConstant
                                    [(IntegerConstant 4 (Integer 4))
                                    (IntegerConstant 3 (Integer 4))
                                    (IntegerConstant 2 (Integer 4))]
                                    (Array
                                        (Integer 4)
                                        [((IntegerConstant 0 (Integer 4))
                                        (IntegerConstant 3 (Integer 4)))]
                                        FixedSizeArray
                                    )
                                    RowMajor
                                )
                                FixedSizeArray
                                DescriptorArray
                                (Array
                                    (Integer 4)
                                    [((IntegerConstant 0 (Integer 4))
                                    (IntegerConstant 3 (Integer 4)))]
                                    DescriptorArray
                                )
                                ()
                            )
                            (Array
                                (Integer 4)
                                [(()
                                ())
                                (()
                                ())
                                (()
                                ())]
                                FixedSizeArray
                            )
                            ()
                        )
                        ()
                    )
                    (=
                        (Var 2 sum1d)
                        (IntegerConstant 0 (Integer 4))
                        ()
                    )
                    (=
                        (Var 2 sum2d)
                        (IntegerConstant 0 (Integer 4))
                        ()
                    )
                    (=
                        (Var 2 i)
                        (IntegerConstant 0 (Integer 4))
                        ()
                    )
                    (WhileLoop
                        ()
                        (IntegerCompare
                            (Var 2 i)
                            Lt
                            (IntegerConstant 5 (Integer 4))
                            (Logical 4)
                            ()
                        )
                        [(=
                            (Var 2 sum1d)
                            (IntegerBinOp
                                (Var 2 sum1d)
                                Add
                                (ArrayItem
                                    (Var 2 arr1d)
                                    [(()
                                    (Var 2 i)
                                    ())]
                                    (Integer 4)
                                    RowMajor
                                    ()
                                )
                                (Integer 4)
                                ()
                            )
                            ()
                        )
                        (=
                            (Var 2 i)
                            (IntegerBinOp
                                (Var 2 i)
                                Add
                                (IntegerConstant 1 (Integer 4))
                                (Integer 4)
                                ()
                            )
                            ()
                        )]
                    )
                    (=
                        (Var 2 j)
                        (IntegerConstant 0 (Integer 4))
                        ()
                    )
                    (WhileLoop
                        ()
                        (IntegerCompare
                            (Var 2 j)
                            Lt
                            (IntegerConstant 4 (Integer 4))
                            (Logical 4)
                            ()
                        )
                        [(=
                            (Var 2 k)
                            (IntegerConstant 0 (Integer 4))
                            ()
                        )
                        (WhileLoop
                            ()
                            (IntegerCompare
                                (Var 2 k)
                                Lt
                                (IntegerConstant 3 (Integer 4))
                                (Logical 4)
                                ()
                            )
                            [(=
                                (Var 2 l)
                                (IntegerConstant 0 (Integer 4))
                                ()
                            )
                            (WhileLoop
                                ()
                                (IntegerCompare
                                    (Var 2 l)
                                    Lt
                                    (IntegerConstant 2 (Integer 4))
                                    (Logical 4)
                                    ()
                                )
                                [(=
                                    (Var 2 sum2d)
                                    (IntegerBinOp
                                        (Var 2 sum2d)
                                        Add
                                        (ArrayItem
                                            (Var 2 arr2d)
                                            [(()
                                            (Var 2 j)
                                            ())
                                            (()
                                            (Var 2 k)
                                            ())
                                            (()
                                            (Var 2 l)
                                            ())]
                                            (Integer 4)
                                            RowMajor
                                            ()
                                        )
                                        (Integer 4)
                                        ()
                                    )
                                    ()
                                )
                                (=
                                    (Var 2 l)
                                    (IntegerBinOp
                                        (Var 2 l)
                                        Add
                                        (IntegerConstant 1 (Integer 4))
                                        (Integer 4)
                                        ()
                                    )
                                    ()
                                )]
                            )
                            (=
                                (Var 2 k)
                                (IntegerBinOp
                                    (Var 2 k)
                                    Add
                                    (IntegerConstant 1 (Integer 4))
                                    (Integer 4)
                                    ()
                                )
                                ()
                            )]
                        )
                        (=
                            (Var 2 j)
                            (IntegerBinOp
                                (Var 2 j)
                                Add
                                (IntegerConstant 1 (Integer 4))
                                (Integer 4)
                                ()
                            )
                            ()
                        )]
                    )
                    (Print
                        [(Var 2 sum1d)
                        (Var 2 sum2d)]
                        ()
                        ()
                    )
                    (=
                        (Var 2 __return_var)
                        (IntegerConstant 0 (Integer 4))
                        ()
                    )
                    (Return)]
                    (Var 2 __return_var)
                    Public
                    .false.
                    .false.
                    ()
                ),
            main_program:
                (Program
                    (SymbolTable
                        3
                        {
                            exit_code:
                                (Variable
                                    3
                                    exit_code
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    main_program
                    [main]
                    [(=
                        (Var 3 exit_code)
                        (FunctionCall
                            1 main
                            1 main
                            []
                            (Integer 4)
                            ()
                            ()
                        )
                        ()
                    )]
                )
        })
    []
)
